_posts/2018-05-29-mono-on-aix-and-ibm-i.md

```diff
@@ -0,0 +1,273 @@
+---
+layout: blog
+title: Introducing Mono on AIX and IBM i
+author: Calvin Buckley
+tags: [porting]
+---
+
+<!-- Edited by Mark Waterbury -->
+
+[![C# REPL running under IBM i](/images/mono-os400.png)](/images/mono-os400.png)
+
+You might have noticed this in the
+[Mono 5.12 release notes](http://www.mono-project.com/docs/about-mono/releases/5.12.0/),
+Mono now includes support for
+[IBM AIX and IBM i](/docs/about-mono/supported-platforms/aix); two very
+different yet (mostly!) compatible operating systems. This post should serve as
+an introduction to this port.
+
+## What does it take to port Mono?
+
+Porting Mono to a new operating system is not as hard as you might think!
+Pretty much the entire world is POSIX compliant these days, and Mono is a large
+yet manageable codebase due to a low number of dependencies, use of plain C99,
+and an emphasis on portability. Most common processor architectures in use are
+supported by the code generator, though more obscure ISAs will have some caveats.
+
+Pretty much all of the work you do will be twiddling `#ifdefs` to accommodate for
+the target platform’s quirks; such as missing or different preprocessor
+definitions and functions, adding the platform to definitions so it is
+supported by core functionality, and occasionally having to tweak the runtime
+or build system to handle when the system does something completely differently
+than others. In the case of AIX and IBM i, I had to do all of these things.
+
+## [Where would I be without IBM?](https://www.youtube.com/watch?v=dYBfdtMAD2Q)
+
+For some background on what needed to happen, we can start by giving some
+background on our target platforms.
+
+Both of our targets run on 64-bit PowerPC processors in big endian mode.
+[Mono does support PowerPC](/docs/about-mono/supported-platforms/powerpc),
```

Comment:
please add a trailing `/` to the link

Reply:
btw. we should link this post from http://www.mono-project.com/docs/about-mono/supported-platforms/aix/ once it's in.